src/main/java/org/junit/runner/Result.java

```diff
@@ -137,7 +137,7 @@ public void testIgnored(Description description) throws Exception {
 
         @Override
         public void testAssumptionFailure(Failure failure) {
-            ignoreCount.getAndIncrement();
+            // do nothing: same as passing (for 4.5; may change in 4.6)
```

Comment:
This is absolutely wrong. We rely on it!
You cannot remove the line. There is no field counter for assumptions and even if it was there, it is a change in behavior in all other frameworks and their integration tests will fail.

Reply:
@Tibor17 I suggest commenting on the bug.